primitives/parachain-inherent/src/client_side.rs

```diff
@@ -181,38 +89,26 @@ fn collect_relay_storage_proof(
 		relay_well_known_keys::hrmp_channels(HrmpChannelId { sender: para_id, recipient })
 	}));
 
-	sp_state_machine::prove_read(relay_parent_state_backend, relevant_keys)
-		.map_err(|e| {
-			tracing::error!(
-				target: LOG_TARGET,
-				relay_parent = ?relay_parent,
-				error = ?e,
-				"Failed to collect required relay chain state storage proof.",
-			)
-		})
-		.ok()
+	relay_chain_interface.prove_read(&relay_parent_block_id, &relevant_keys).ok()?
```

Comment:
Weird that you get here a result and convert it to Option and below you just return Option directly.

Reply:
I made this consistent now, If an error is possible, I return `Result` from the interface